portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/fragments/reset-password/reset-password.js

```diff
@@ -24,6 +24,25 @@ function updatePassword(username, oldPassword, newPassword, domain) {
         return {success: true, message: "You have successfully updated the password"};
     } catch (e) {
         var message = e.message;
+        var errorCode = -1;
+        var cause = e.getCause();
+        if (cause != null) {
+            //the exceptions thrown by the actual osgi service method is wrapped inside a InvocationTargetException.
+            if (cause instanceof java.lang.reflect.InvocationTargetException) {
+                message = cause.getTargetException().message;
+        		errorCode = cause.getTargetException().getErrorCode();
+
+        		if (errorCode === "1000") {
+                    var count = callOSGiService("org.wso2.is.portal.user.client.api.IdentityStoreClientService",
+                            "getHistoryCount", []);
+                    message = "Password You entered has already been used in last " + count + "Attempts";
```

Comment:
I assume this error shown to the end user. In that case this need to be properly internationalized. Better to have message as "Entered password has already been used in last x days."  Attention capital, simple letter usage too.